template.yaml

```diff
@@ -370,6 +723,342 @@ Resources:
+            options: { } # workaround to get `sam validate` to work
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: "openAPI/api.yaml"
+      Description: "Regional REST API that provides method to interact with the UserIdentity function"
+      StageName: !Ref ApiStageName
+      AccessLogSetting:
+        Format: >-
+          {
+          "requestId": "$context.requestId",
+          "requestTime": "$context.requestTime",
+          "requestTimeEpoch": "$context.requestTimeEpoch",
+          "httpMethod": "$context.httpMethod",
+          "domainName": "$context.domainName",
+          "path": "$context.path",
+          "routeKey": "$context.routeKey",
+          "protocol": "$context.protocol",
+          "status": "$context.status",
+          "responseLength": "$context.responseLength",
+          "responseLatency": "$context.responseLatency",
+          "authorizerLatency": "$context.authorizer.latency",
+          "integrationStatus": "$context.integrationStatus",
+          "integrationLatency": "$context.integrationLatency",
+          "integrationErrorMessage": "$context.integrationErrorMessage",
+          "errorMessage": "$context.error.message",
+          "errorResponseType": "$context.error.responseType",
+          "validationErrorString": "$context.error.validationErrorString",
+          "xrayTraceId": "$context.xrayTraceId",
+          "apiKeyId": "$context.identity.apiKeyId"
+          }
+        DestinationArn: !GetAtt APIGatewayAccessLogGroup.Arn
+      CacheClusterEnabled: false
+      TracingEnabled: true
+      EndpointConfiguration:
+        Type: REGIONAL
+      DisableExecuteApiEndpoint: false
+      Tags:
+        CheckovRulesToSkip: CKV_AWS_120
+        Product: !Ref ProductTagValue
+        System: !Ref SystemTagValue
+        Environment: !Ref Environment
+        Owner: !Ref OwnerTagValue
+        Source: !Ref SourceTagValue
+        FMSRegionalPolicy: false
+        CustomPolicy: true
+
+  APIGatewayAccessLogGroup:
+    Type: AWS::Logs::LogGroup
+    Condition: IsDev
+    Properties:
+      LogGroupName: !Sub "/aws/apigateway/${RestApi}/${ApiStageName}"
+      RetentionInDays: 30
+      KmsKeyId: !GetAtt CloudWatchEncryptionKey.Arn
+      Tags:
+        - Key: Product
+          Value: !Ref ProductTagValue
+        - Key: System
+          Value: !Ref SystemTagValue
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Name
+          Value: !Sub "/aws/apigateway/${RestApi}/${ApiStageName}"
+
+  ########################
+  # IPV Core Usage Plans #
+  ########################
+  IpvCoreApiUsagePlan:
+    Type: AWS::ApiGateway::UsagePlan
+    Condition: IsDev
+    DependsOn:
+      - RestApiStage # This resource is created as part of the RestApi resource
+    Properties:
+      UsagePlanName: "IPV Core"
```

Comment:
Can we include stack name in the name please, it can very confusing in `dev` where there are multiple stacks deployed to see which usage plan belongs to which stack.